### PR TITLE
store/mockstore/unistore: refine and add more mpp tests

### PR DESCRIPTION
**Pull request by hanfei1991**
### What problem does this PR solve?

Problem Summary: support mock mpp execution without pointing a table.

### What is changed and how it works?

What's Changed: unistore is refined to call mpp execution directly.

How it Works:

### Related changes

- PR to update `unistore`: https://github.com/ngaut/unistore/pull/436

### Check List 

Tests 

- Unit test

### Release note 

-  refine and add more mpp tests

### CODE REVIEW

File: store/mockstore/unistore/cophandler/mpp_exec.go [hunk 1]
```diff
@@ -153,16 +155,44 @@ func (e *exchSenderExec) next() (*chunk.Chunk, error) {
...
 			return nil, nil
 		} else if chk != nil {
-			for _, tunnel := range e.tunnels {
-				tipbChunks, err := e.toTiPBChunk(chk)
-				if err != nil {
-					for _, tunnel := range e.tunnels {
-						tunnel.ErrCh <- err
+			if e.exchangeTp == tipb.ExchangeType_Hash {
+				rows := chk.NumRows()
+				targetChunks := make([]*chunk.Chunk, 0)
```
Comment (ngaut) [inline store/mockstore/unistore/cophandler/mpp_exec.go:160]: Make with cap by using length of targetChunks

Comment (hanfei1991) [reply store/mockstore/unistore/cophandler/mpp_exec.go]: fixed

**Review by qw4990 — COMMENTED** [review]
/LGTM

File: executor/tiflash_test.go [hunk 2]
```diff
@@ -138,4 +137,9 @@ func (s *tiflashTestSuite) TestMppExecution(c *C) {
 	tk.MustQuery("select avg(t1.a) from t1 , t where t1.a = t.a").Check(testkit.Rows("2.0000"))
 	// test proj and selection
 	tk.MustQuery("select count(*) from (select a * 2 as a from t1) t1 , (select b + 4 as a from t)t where t1.a = t.a").Check(testkit.Rows("3"))
+
+	// test shuffle hash join.
+	tk.MustExec("set @@session.tidb_broadcast_join_threshold_size=1")
+	tk.MustQuery("select count(*) from t1 , t where t1.a = t.a").Check(testkit.Rows("3"))
+	tk.MustQuery("select count(*) from t1 , t, t2 where t1.a = t.a and t2.a = t.a").Check(testkit.Rows("3"))
```
Comment (fzhedu) [inline executor/tiflash_test.go:144]: ... obtained string = "[tikv:1]unsupported executor type TypeSelection"
... expected string = ""
... sql:select b from t3 where a >1, 

can this SQL run correctly? which errors before several days.

File: store/mockstore/unistore/cophandler/mpp_exec.go [hunk 3]
```diff
@@ -153,16 +155,44 @@ func (e *exchSenderExec) next() (*chunk.Chunk, error) {
...
+							tunnel.DataCh <- &tipbChunk
+						}
 					}
-					return nil, nil
 				}
-				for _, tipbChunk := range tipbChunks {
-					tunnel.DataCh <- &tipbChunk
+			} else {
+				for _, tunnel := range e.tunnels {
+					tipbChunks, err := e.toTiPBChunk(chk)
```
Comment (fzhedu) [inline store/mockstore/unistore/cophandler/mpp_exec.go:186]: would the chunk be empty?

File: store/mockstore/unistore/cophandler/mpp.go [hunk 4]
```diff
@@ -327,46 +338,6 @@ type MPPTaskHandler struct {
 	Err    error
 }
 
-// HandleMPPDispatch handle DispatchTaskRequest
```
Comment (fzhedu) [inline store/mockstore/unistore/cophandler/mpp.go:330]: Is this function not called previously?

File: executor/tiflash_test.go (hunk 2 above)
Comment (hanfei1991) [reply executor/tiflash_test.go:144]: Of course

File: store/mockstore/unistore/cophandler/mpp_exec.go (hunk 3 above)
Comment (hanfei1991) [reply store/mockstore/unistore/cophandler/mpp_exec.go:186]: It's not supposed to be.

File: store/mockstore/unistore/cophandler/mpp.go (hunk 4 above)
Comment (hanfei1991) [reply store/mockstore/unistore/cophandler/mpp.go:330]: It did be called. For now I move it to the unistore and polish it.

**Review by fzhedu — APPROVED** [review]
LGTM

**Review by qw4990 — APPROVED** [review]

**Pull request merged**